catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/ShareableMetacardAccessPlugin.java

```diff
@@ -66,25 +67,25 @@ public CreateRequest processPreCreate(CreateRequest input) throws StopProcessing
   public UpdateRequest processPreUpdate(
       UpdateRequest input, Map<String, Metacard> existingMetacards) throws StopProcessingException {
 
-    Function<WorkspaceMetacardImpl, WorkspaceMetacardImpl> previous =
-        (update) -> WorkspaceMetacardImpl.from(existingMetacards.get(update.getId()));
+    Function<ShareableMetacardImpl, ShareableMetacardImpl> previous =
+        (update) -> ShareableMetacardImpl.from(existingMetacards.get(update.getId()));
 
     Set<String> notOwners =
         input
             .getUpdates()
             .stream()
             .map(Map.Entry::getValue)
-            .filter(WorkspaceMetacardImpl::isWorkspaceMetacard)
-            .map(WorkspaceMetacardImpl::from)
+            .filter(ShareableMetacardImpl::canShare)
+            .map(ShareableMetacardImpl::from)
```

Comment:
If we decided to throw an exception in `from` it looks like this path will still execute as expected, since filtering is done on the shareability check.